torchdp/autograd_grad_sample.py

```diff
@@ -31,29 +31,39 @@
 # work-around for https://github.com/pytorch/pytorch/issues/25723
 _hooks_disabled: bool = False
 
-# global switch to catch double backprop errors on Hessian computation
-_enforce_fresh_backprop: bool = False
 
-
-def add_hooks(model: nn.Module) -> None:
+def add_hooks(model: nn.Module, loss_type: str = "mean", batch_dim: int = 0) -> None:
```

Comment:
nit: we could pass an IntEnum instead.

Reply:
Some parts of the code (e.g., parts of the clipper) silently assume that loss_type=mean, and batch_dim=0. 
Is this generality really warranted here? E.g., I don't see a compelling reason why the batch dimension wouldn't be zero.

Reply:
you are right, it is probably unnecessary  but let's do that in another diff.

Reply:
Ok! then I'll leave this as is for now